### PR TITLE
HW1 Completed Pull Request

### PR DESCRIPTION
**Pull request by NAlexH2**
This pr is part of the first homework an demonstrates the following:

1. Completion of all Flexbox Froggy exercises. 
    * I neglected to start a new branch before completing these to add to the PR, but I don't think they are as important as the 2nd exercise.
3. `02-palindrome.js` implementation following Airbnb guidelines
4. `02-palindrome.html` additions to support bootstrap CSS styling and accessibility features for the label/input.

**Note:** I will assign my group members once they have accepted the invitation.

### CODE REVIEW

File: hw1/02-palindrome.js [hunk 1]
```diff
@@ -1,3 +1,46 @@
-const elem = document.querySelector('input');
+const inputElem = document.querySelector("input");
+const NOT_A_NUMBER = "Not a number.";
+const POSITIVES_ONLY = "Input cannot be less than 0.";
```
Comment (PhatCash) [inline hw1/02-palindrome.js:3]: The Airbnb documentation mentions using single quotes for strings **Rule 6.1**. However, as mentioned in class, prettier automatically saves them as double quotes and the professor isn't exactly looking for this precise integration.

File: hw1/02-palindrome.js [hunk 2]
```diff
@@ -1,3 +1,46 @@
...
+    } else if (Number(receivedInput) < 0) {
+        dangerElem(POSITIVES_ONLY);
+        return;
+    } else if (receivedInput !== inputReversed) {
+        dangerElem(NOT_PALINDROME);
+        return;
+    }
+
+    elemRes = successElem(IS_PALINDROME);
+}
```
Comment (PhatCash) [inline hw1/02-palindrome.js:46]: **Rule 19.5** mentioned ending a file with a single newline character. Again, I don't know how important this is like the rule of using double or single quotes.

**Review by PhatCash — APPROVED** [review]
Code review is complete. Nothing major.

File: hw1/02-palindrome.js (hunk 1 above)
Comment (NAlexH2) [reply hw1/02-palindrome.js:3]: That's why I brought it up. And the rule doesn't even state why but the ES standard just says "be consistent with the usage" for how it's being used. The single vs double is probably just the standard they (Airbnb) wanted to use in their code base. Thanks for taking a look.

File: hw1/02-palindrome.js (hunk 2 above)
Comment (NAlexH2) [reply hw1/02-palindrome.js:46]: Interesting, I'll check my source on my home pc but on my laptop that new line is present. Even viewing the file directly on gh shows it's missing. Wonder what's going on there. Though it's not a huge issue.

Comment (NAlexH2) [reply hw1/02-palindrome.js:46]: @PhatCash  Looks like if you don't have one GH will alert with a big red circle, but if you do, it simply ignores it and doesn't put an alert. Looks like it just eats that last line. See Baylie's files to see what I'm talking about and compare her files EOF line vs mine.

Comment (PhatCash) [reply hw1/02-palindrome.js:46]: Ah. I see what you're referring to. I have not seen that before, so it's interesting.

**Comment by baylie-r-g** [comment]
Hey Alex!
This is looking good! I 1000% did not remember that the color classes text-danger and text-success existed and I just did .style.color instead. I find the lack of col- classes interesting, but I can see how margin classes can make the display more consistent. All around well done!

**Review by baylie-r-g — APPROVED** [review]
See the comment I left. I was unsure how to do this process lol

File: hw1/02-palindrome.js [hunk 3]
```diff
@@ -1,3 +1,46 @@
-const elem = document.querySelector('input');
+const inputElem = document.querySelector("input");
+const NOT_A_NUMBER = "Not a number.";
+const POSITIVES_ONLY = "Input cannot be less than 0.";
+const NOT_PALINDROME = "No. Try again";
+const IS_PALINDROME = "Yes! This number is a palindrome!";
```
Comment (glatimer) [inline hw1/02-palindrome.js:5]: Declaring your expected outputs in the beginning is an organized way to start!

File: hw1/02-palindrome.js [hunk 4]
```diff
@@ -1,3 +1,46 @@
...
 
-elem.addEventListener('input', handleInput);
+inputElem.addEventListener("input", handleInput);
+
+function inputVerify(input) {
+    const verifyResult = isNaN(input) === true || input.length === 0;
+    return verifyResult;
+}
+
```
Comment (glatimer) [inline hw1/02-palindrome.js:13]: It's not a huge program so this is less important, but adding some comments to your functions would be beneficial.

File: hw1/02-palindrome.js [hunk 5]
```diff
@@ -1,3 +1,46 @@
...
+    modElem.classList = "";
+    modElem.classList.add("text-success");
+    modElem.textContent = message;
+    return;
+}
+
+function handleInput() {
+    const receivedInput = inputElem.value;
+    const inputReversed = Array.from(receivedInput).reverse().join("");
+
```
Comment (glatimer) [inline hw1/02-palindrome.js:33]: Clever idea to make the user input into an array for later manipulation/reversal check.

**Review by glatimer — APPROVED** [review]

**Pull request merged**